### PR TITLE
Update FederatedTypeConfig to be namespace-scoped

### PR DESCRIPTION
**Pull request by marun**
This is the next step in allowing the fedv2 control plane to run in an arbitrary namespace (#197).  As part of this PR I updated the CRD e2e test to support targeting multiple clusters to ensure this change can be tested against the default managed fixture which uses 2 clusters.

### CODE REVIEW

File: pkg/apis/core/v1alpha1/federatedtypeconfig_types_test.go [hunk 1]
```diff
@@ -51,7 +51,7 @@ var _ = Describe("FederatedTypeConfig", func() {
 	Describe("when sending a storage request", func() {
 		Context("for a valid config", func() {
 			It("should provide CRUD access to the object", func() {
-				client = cs.CoreV1alpha1().FederatedTypeConfigs()
+				client = cs.CoreV1alpha1().FederatedTypeConfigs("foo")
```
Comment (gyliu513) [inline pkg/apis/core/v1alpha1/federatedtypeconfig_types_test.go]: Just a question here, why did the `kubebuilder` did not update the test file automatically when change from cluster scope to namespace scope?

Comment (marun) [reply pkg/apis/core/v1alpha1/federatedtypeconfig_types_test.go]: kubebuilder doesn't maintain generated files.

Comment (gyliu513) [reply pkg/apis/core/v1alpha1/federatedtypeconfig_types_test.go]: I see, thanks @marun

Comment (font) [reply pkg/apis/core/v1alpha1/federatedtypeconfig_types_test.go]: These files are generated as scaffolding when you first create a resource type, but no longer maintained when regenerating code using kubebuilder. Perhaps that could be a feature in kubebuilder, but often times you wouldn't want to regenerate these unit tests if you've heavily modified the unit tests after the fact.

File: test/e2e/crud.go [hunk 2]
```diff
@@ -92,68 +91,76 @@ var _ = Describe("Federated types", func() {
 
 			// TODO(marun) Is there a better way to create crd's from code?
 
-			crdKind := "FedTestCrd"
+			targetCrdKind := "FedTestCrd"
+			targetCrd := newTestCrd(tl, targetCrdKind)
```
[the remark this answers is not in the document]
Comment (marun) [reply test/e2e/crud.go]: The goal of this test was to validate federation of a CRD target, since all the existing CRUD tests target first-class api types.  I didn't test overrides because we are already validating CRD-based overrides in most of the CRUD tests.  What benefit(s) do you see in adding a check for overrides to this test?

File: test/e2e/crud.go [hunk 3]
```diff
@@ -92,68 +91,76 @@ var _ = Describe("Federated types", func() {
 
 			// TODO(marun) Is there a better way to create crd's from code?
 
-			crdKind := "FedTestCrd"
+			targetCrdKind := "FedTestCrd"
```
[the remark this answers is not in the document]
Comment (marun) [reply test/e2e/crud.go]: There are only 2 nested describes in this file, so I'm not sure what you mean.  How would you propose the tests be structured?

[the remark this answers is not in the document]
Comment (font) [reply test/e2e/crud.go]: It's more of a combination of all the describe containers, it specs, and anonymous functions that seems to be hurting readability at first glance. Once you understand how it's broken up it makes sense. I haven't thought much about it and don't know if there is a Ginkgo best practice for this. Some thoughts that come to mind:

- It may be possible to avoid too many nested anonymous functions and replace the anonymous functions with identifiers for those functions defined elsewhere. However, this could be problematic with the use of function closures.
- The 2 nested describes appear to be sharing the federation framework, e2e logger, and namespaceTypeConfig. Perhaps there's a better way to share this context amongst the tests.
- I think the CRD Describe container is probably the one that is getting a bit long. I think focusing on this would probably help the most such as refactoring it into separate functions that can possibly be re-used. Whenever we start generating the CRDs, it can probably be refactored too.

Anyway, not sure how valid any of those ideas are without investigating further. For now we have higher priority stuff.

File: test/e2e/crud.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (marun) [reply test/e2e/crud.go]: Validating FederatedTypeConfig with crd overrides is already being done when e2e crud is targeting unmanaged, though. There is no difference between a FederatedTypeConfig loaded from config and one a user would define (since fed primitives are CRD in both cases).  This test validates that the sync controller is capable of working with a CRD as the target, which is a corner case the other crud tests don't cover.
